autode/pes/pes_nd.py

```diff
@@ -0,0 +1,839 @@
+        Arguments:
+            atom_idxs: Atom indices
+            value:
+
+        Returns:
+            (autode.pes.pes_nd._Distances1D):
+        """
+
+        if len(value) == 2:
+
+            if self._species is None:
+                raise ValueError('Cannot determine initial point without '
+                                 'a defined species')
+
+            # Have a pair, the final distance and either the number of steps
+            # or the step size
+            r_init, r_final = self._species.distance(*atom_idxs), value[0]
+
+        elif len(value) == 3:
+            # A triple also defines the initial distance
+            r_init, r_final = value[0], value[1]
+
+        else:
+            raise ValueError(f'Cannot interpret *{value}* as a final '
+                             f'distance and number of steps or step size')
+
+        if isinstance(value[-1], int):
+            # Integer values must be a number of steps
+            num = value[-1]
+
+        elif isinstance(value[-1], float):
+            num = int(round(abs((r_final - r_init) / value[-1]))) + 1
+
+            if not self._allow_rounding_of_stepsize:
+                dr = np.sign(r_final - r_init) * abs(value[-1]) * (num - 1)
+                r_final = r_init + dr
+
+        else:
+            raise ValueError(f'Uninterpretable type: {type(value)}')
+
+        if num <= 1:
+            raise ValueError(f'Unsupported number of steps: {num}')
+
+        return _Distances1D(np.linspace(r_init, r_final, num=num),
+                            atom_idxs=atom_idxs)
+
+    def __eq__(self, other):
+        """Equality of two _ListDistances1D instances"""
+        return isinstance(other, _ListDistances1D) and super().__eq__(other)
+
+
+class _Distances1D(ValueArray):
+
+    implemented_units = [ang]
+
+    def __new__(cls,
+                input_array: Union[np.ndarray, Sequence],
+                atom_idxs:   Tuple[int, int]):
+        """
+        Create an array of distances in a single dimension, with associated
+        atom indices, indexed from 0
+
+        -----------------------------------------------------------------------
+        Arguments:
+            input_array: Array of distances e.g. [1.0, 1.1, 1.2] in Å
+
+            atom_idxs: Indices of the atoms involved in this distance
+                       e.g. (0, 1)
+        """
+        arr = super().__new__(cls, input_array=input_array, units=ang)
+
+        if len(atom_idxs) != 2:
+            raise ValueError(f'Indices must be a 2-tuple. Had: {atom_idxs}')
+
+        i, j = atom_idxs
+        if not (isinstance(i, int) and isinstance(j, int)):
+            raise ValueError(f'Atom indices must be integers. Had: {i}, {j}')
+
+        if i < 0 or j < 0:
+            raise ValueError(f'Atom indices must be >0: Had {i}, {j}')
+
+        arr.atom_idxs = atom_idxs
+        return arr
+
+    @property
+    def min(self) -> float:
+        """
```

Comment:
This docstring is fundamentally pointless

Reply:
this is true